tests/asahirov/test_homework_7.py

```diff
@@ -0,0 +1,191 @@
+import random
+from operator import gt, lt, le, ge
+import time
+
+"""
+Create decorator which would remember answers,
+and if answer was given, do not make a call to a function
+"""
+
+
+def remember_answer(func):
+    answer = {}
+
+    def remember(*args):
+        if args in answer:
+            return answer[args]
+        else:
+            result = func(*args)
+            answer[args] = result
+            return result
+
+    return remember
+
+
+@remember_answer
+def fibo(n: int):
+    """
+    Count fibonache numbers, where next is sum of 2 prior
+    1, 2, 3, 5, 8, 13, 21, ...
+    """
+    if n < 3:
+        return [0, 1, 2][n]
+    return fibo(n - 1) + fibo(n - 2)
+
+
+def test_remember_answer_decorator():
+    for i in range(10):
+        print(fibo(i))
+
+
+# ----------------------------------------------------------------------------------
+"""
+Create decorator, which will retry inner function multiple times untill it passes
+"""
+
+
+def retry_decorator(retries: int = 5):
+    def retry(func):
+        def wrapper():
+            for i in range(retries):
+                try:
+                    return func()
```

Comment:
you limited decorator to work only with functions which do not expect any input params
also you limited it by 1 single error type, which is bad

Reply:
use *args, *kwargs.      and Exception,
also optionaly user could pass tuple of errors  he would like to intercept